infoschema/builder.go

```diff
@@ -73,7 +73,7 @@ func (b *Builder) ApplyDiff(m *meta.Meta, diff *model.SchemaDiff) ([]int64, erro
 	// We try to reuse the old allocator, so the cached auto ID can be reused.
 	var alloc autoid.Allocator
 	if tableIDIsValid(oldTableID) {
-		if oldTableID == newTableID {
+		if oldTableID == newTableID && diff.Type != model.ActionRenameTable {
```

Comment:
We can still reuse the allocator for rename table, just add a method `SetDBID`.

Reply:
But, it needs to add an interface to Allocator.

Reply:
Alright, skip auto ID when rename table is acceptable.